WooCommerce/Classes/ViewRelated/ReusableViews/LedgerTableViewCell.swift

```diff
@@ -36,66 +36,81 @@ final class LedgerTableViewCell: UITableViewCell {
-        discountValue.text = viewModel.discountValue
-        discountView.isHidden = viewModel.shouldHideDiscount
-
-        feesLabel.text = Titles.feesLabel
-        feesValue.text = viewModel.feesValue
-        feesView.isHidden = viewModel.shouldHideFees
-
-        shippingLabel.text = Titles.shippingLabel
-        shippingValue.text = viewModel.shippingValue
-
-        taxesLabel.text = Titles.taxesLabel
-        taxesValue.text = viewModel.taxesValue
-        taxesView.isHidden = viewModel.shouldHideTaxes
-
-        totalLabel.text = Titles.totalLabel
-        totalValue.text = viewModel.totalValue
-
-        accessibilityElements = [subtotalLabel as Any,
-                                 subtotalValue as Any,
-                                 discountLabel as Any,
-                                 discountValue as Any,
-                                 feesLabel as Any,
-                                 feesValue as Any,
-                                 shippingLabel as Any,
-                                 shippingValue as Any,
-                                 taxesLabel as Any,
-                                 taxesValue as Any,
-                                 totalLabel as Any,
-                                 totalValue as Any
-                                ]
+        configureSubtotal(label: Titles.subtotalLabel, value: viewModel.subtotalValue)
+        configureDiscount(label: viewModel.discountText, value: viewModel.discountValue, hidden: viewModel.shouldHideDiscount)
+        configureFees(label: Titles.feesLabel, value: viewModel.feesValue, hidden: viewModel.shouldHideFees)
+        configureShipping(label: Titles.shippingLabel, value: viewModel.shippingValue, hidden: false)
+        configureTaxes(label: Titles.taxesLabel, value: viewModel.taxesValue, hidden: viewModel.shouldHideTaxes)
+        configureTotal(label: Titles.totalLabel, value: viewModel.totalValue)
+        configureAccessibility()
     }
 
     /// Configure a refund details "table"
     ///
     func configure(with viewModel: RefundDetailsViewModel) {
-        subtotalLabel.text = Titles.subtotal
-        subtotalValue.text = viewModel.itemSubtotal
+        configureSubtotal(label: Titles.subtotal, value: viewModel.itemSubtotal)
+        configureDiscount(label: nil, value: nil, hidden: true)
+        configureFees(label: nil, value: nil, hidden: true)
+        configureShipping(label: nil, value: nil, hidden: true)
+        configureTaxes(label: Titles.tax, value: viewModel.taxSubtotal, hidden: taxesValue == nil)
+        configureTotal(label: Titles.productsRefund, value: viewModel.productsRefund)
+        configureAccessibility()
+    }
+
+    private func configureSubtotal(label: String, value: String) {
+        subtotalLabel.text = label
+        subtotalValue.text = value
+        subtotalView.accessibilityLabel = "\(label) \(value)"
+    }
+
+    private func configureDiscount(label: String?, value: String?, hidden: Bool) {
```

Comment:
Not a big deal, but maybe the body of these three methods could be extracted to a separate method, and pass the label as a parameter?

Reply:
@ctarda See latest changes... I'd considered having a shared method before I PRed, but passing in the three views each time seemed like it would make the method call unwieldy and reduce readability.

You prompted me to revisit it, and I tried grouping the views to avoid this issue. I quite like the nested struct here, since it keeps things readable at the call site. What do you think? 

Thanks for prompting me to improve it.